frontend/src/views/setting/panel/index.vue

```diff
@@ -438,48 +438,50 @@ const onChangeThemeColor = () => {
     themeColorRef.value.acceptParams({ themeColor: themeColor, theme: globalStore.themeConfig.theme });
 };
 
-const onSave = async (key: string, val: any) => {
-    loading.value = true;
-    if (key === 'Language') {
-        i18n.locale.value = val;
-        globalStore.updateLanguage(val);
-    }
-    if (key === 'Theme') {
-        globalStore.themeConfig.theme = val;
-        switchTheme();
-        if (globalStore.isProductPro) {
-            await updateXpackSettingByKey('Theme', val);
-            let color: string;
-            const themeColor: ThemeColor = JSON.parse(globalStore.themeConfig.themeColor);
-            if (val === 'auto') {
-                const prefersDark = window.matchMedia('(prefers-color-scheme: dark)');
-                color = prefersDark.matches ? themeColor.dark : themeColor.light;
-            } else {
-                color = val === 'dark' ? themeColor.dark : themeColor.light;
-            }
-            globalStore.themeConfig.primary = color;
-            setPrimaryColor(color);
+const handleThemeChange = async (val: string) => {
+    globalStore.themeConfig.theme = val;
+    switchTheme();
+    if (globalStore.isProductPro) {
+        await updateXpackSettingByKey('Theme', val);
+        let color: string;
+        const themeColor: ThemeColor = JSON.parse(globalStore.themeConfig.themeColor);
+        if (val === 'auto') {
+            const prefersDark = window.matchMedia('(prefers-color-scheme: dark)');
+            color = prefersDark.matches ? themeColor.dark : themeColor.light;
+        } else {
+            color = val === 'dark' ? themeColor.dark : themeColor.light;
         }
+        globalStore.themeConfig.primary = color;
+        setPrimaryColor(color);
     }
-    if (key === 'MenuTabs') {
-        globalStore.setOpenMenuTabs(val === 'enable');
-    }
+};
+
+const onSave = async (key: string, val: any) => {
+    loading.value = true;
     let param = {
         key: key,
         value: val + '',
     };
-    await updateSetting(param)
-        .then(async () => {
-            if (param.key === 'Language') {
-                location.reload();
-            }
-            loading.value = false;
-            MsgSuccess(i18n.t('commons.msg.operationSuccess'));
-            await search();
-        })
-        .catch(() => {
-            loading.value = false;
-        });
+    try {
+        await updateSetting(param);
+        if (key === 'Language') {
+            i18n.locale.value = val;
+            globalStore.updateLanguage(val);
+            location.reload();
+        }
+
+        if (key === 'Theme') {
+            await handleThemeChange(val);
+        }
+        if (key === 'MenuTabs') {
+            globalStore.setOpenMenuTabs(val === 'enable');
+        }
+        MsgSuccess(i18n.t('commons.msg.operationSuccess'));
+        await search();
+    } catch (error) {
+    } finally {
+        loading.value = false;
+    }
 };
 
 onMounted(() => {
```

Comment:
There appear to be some changes in the structure of the `onSave` function, particularly around how it handles different keys ('Language', 'Theme', 'MenuTabs'). Here are some improvements and suggestions:

### Key Improvements:
1. **Unified Logic for Theme Handling**:
   - Combine the logic related to `updateXpackSettingByKey`, switching themes, updating primary color, and setting menu tabs into a single method (`handleThemeChange`). This reduces duplication and makes the code cleaner.

2. **Error Handling**:
   - Added a `catch` block within the promise chain to ensure that any errors during processing do not cause the flow to crash.

3. **Loading Value Management**:
   - Moved the finalization of the loading state outside both the `then` and `catch` blocks. The loading indicator should only be turned off on success or error.

4. **Return Statement**:
   - Since you mentioned checking for specific types (e.g., `"string"`), it would make sense to include these checks at the start of the function to prevent unnecessary operations. However, based on your provided snippet, this is assumed handled elsewhere.

5. **Code Readability**:
   - Improved indentation and spacing for better readability.

Here's an updated version of the `onSave` function with these considerations applied:

```javascript
const handleThemeChange = async (val: string): Promise<void> => {
    globalStore.themeConfig.theme = val;
    switchTheme();
    if (globalStore.isProductPro) {
        await updateXpackSettingByKey('Theme', val);
        let color: string;
        const themeColor: ThemeColor = JSON.parse(globalStore.themeConfig.themeColor);
        if (val === 'auto') {
            const prefersDark = window.matchMedia('(prefers-color-scheme: dark)');
            color = prefersDark.matches ? themeColor.dark : themeColor.light;
        } else {
            color = val === 'dark' ? themeColor.dark : themeColor.light;
        }
        globalStore.themeConfig.primary = color;
        setPrimaryColor(color);
    }
};

async function onSave(key: string, val: any): Promise<void> {
    loading.value = true;

    // Check if the value is a string before proceeding further
    if (typeof val !== 'string') return;

    let param: { key: string; value: string | number } = {
        key,
        value: val.toString(),
    };

    try {
        await updateSetting(param);

        if (key === 'Language') {
            i18n.locale.value = val;
            globalStore.updateLanguage(val);
            location.reload();
        }

        if (key === 'Theme') {
            await handleThemeChange(val);
        }

        if (key === 'MenuTabs') {
            globalStore.setOpenMenuTabs(val === 'enable');
        }

        Message.success(i18n.t('commons.msg.operationSuccess'));
        await search();
    } catch (error) {
        console.error('An error occurred:', error); // Add logging here or use a more appropriate error handling approach
    }

    finally {
        loading.value = false;
    }
}

onMounted(() => {
```

Please note that I've included type-checking for strings to handle unexpected data types early in the process. If you prefer logging instead of catching, feel free to adjust accordingly. Additionally, ensure that all external functions used (like `location.reload()`, `setPrimaryColor`, `search()` etc.) exist and perform their intended actions correctly.